### PR TITLE
chore: Pre-build DevWorkspace templates for devfile v2

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

### What does this PR do?
Generate DW templates at build time and put them into `/usr/local/apache2/htdocs/devfiles/v2/che-theia/` folder in the che-devfile-registry image.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-2500

### CODE REVIEW

**Review by svor — APPROVED** [review]
LGTM

File: dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  devfile_url=$(grep "\"v2\":" "${dir}"meta.yaml) || :
+  if [ -n "$devfile_url" ]; then
+    devfile_url=${devfile_url##*\"v2\": \"}
+    devfile_url=${devfile_url%/}
+    devfile_url=${devfile_url%\"*}
+    devfile_repo=${devfile_url%/tree*}
+    name=$(basename "${devfile_repo}")
+
+    npm_config_yes=true npx @eclipse-che/che-theia-devworkspace-handler --devfile-url:"${devfile_url}" \
+    --output-file:"${dir}"/devworkspace-che-theia-next.yaml \
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh:29]: same question from https://github.com/redhat-developer/codeready-workspaces-images/pull/156/commits ... why are we building a @eclipse-che/che-theia-devworkspace-handler next ? how is this usable in CRW? 

also your use of quotes in 
```
    --project."${name}={{ INTERNAL_URL }}/resources/v2/${name}.zip"
```

looks strange. Would a better markup be... 
```
    "--project.${name}"="{{ INTERNAL_URL }}/resources/v2/${name}.zip"

```

Comment (svor) [inline dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh:29]: Let's have only **latest** in CRW

Comment (vinokurig) [reply dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh]: > same question from https://github.com/redhat-developer/codeready-workspaces-images/pull/156/commits ... why are we building a @eclipse-che/che-theia-devworkspace-handler next ? how is this usable in CRW?

Removed the generation of `devworkspace-che-theia-next.yaml`
> also your use of quotes in 
>```
>    --project."${name}={{ INTERNAL_URL }}/resources/v2/${name}.zip"
>```
>
>looks strange. Would a better markup be... 
>```
>    "--project.${name}"="{{ INTERNAL_URL }}/resources/v2/${name}.zip"
>```

fixed

Comment (vinokurig) [reply dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh]: done

**Comment by svor** [comment]
@vinokurig could you please provide an example of the resulted generated DevWorkspace template

**Comment by vinokurig** [comment]
@svor 
> @vinokurig could you please provide an example of the resulted generated DevWorkspace template


```
apiVersion: workspace.devfile.io/v1alpha2
kind: DevWorkspaceTemplate
metadata:
  name: theia-ide-quarkus-quickstart
spec:
  commands:
    - id: init-container-command
      apply:
        component: remote-runtime-injector
  events:
    preStart:
      - init-container-command
  components:
    - name: theia-ide
      container:
        image: >-
          quay.io/eclipse/che-theia@sha256:ebcc893364c233f4ebbaef61c6b2027c2aabecf05844f6151837dbb13193480d
        env:
          - name: THEIA_PLUGINS
            value: local-dir:///plugins
          - name: HOSTED_PLUGIN_HOSTNAME
            value: 0.0.0.0
          - name: HOSTED_PLUGIN_PORT
            value: '3130'
          - name: THEIA_HOST
            value: 127.0.0.1
        volumeMounts:
          - name: plugins
            path: /plugins
          - name: theia-local
            path: /home/theia/.theia
        mountSources: true
        memoryLimit: 512M
        cpuLimit: 1500m
        cpuRequest: 100m
        endpoints:
          - name: theia
            attributes:
              type: main
              cookiesAuthEnabled: true
              discoverable: false
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: webviews
            attributes:
              type: webview
              cookiesAuthEnabled: true
              discoverable: false
              unique: true
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: mini-browser
            attributes:
              type: mini-browser
              cookiesAuthEnabled: true
              discoverable: false
              unique: true
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: theia-dev
            attributes:
              type: ide-dev
              discoverable: false
              urlRewriteSupported: true
            targetPort: 3130
            exposure: public
            protocol: http
          - name: theia-redirect-1
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13131
            exposure: public
            protocol: http
          - name: theia-redirect-2
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13132
            exposure: public
            protocol: http
          - name: theia-redirect-3
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13133
            exposure: public
            protocol: http
          - name: terminal
            attributes:
              type: collocated-terminal
              discoverable: false
              cookiesAuthEnabled: true
              urlRewriteSupported: true
            targetPort: 3333
            exposure: public
            secure: false
            protocol: wss
      attributes:
        app.kubernetes.io/component: che-theia
        app.kubernetes.io/part-of: che-theia.eclipse.org
    - name: plugins
      volume: {}
    - name: theia-local
      volume: {}
    - name: che-machine-exec
      container:
        image: >-
          quay.io/eclipse/che-machine-exec@sha256:2ecefe3a93c1ce35b0d56beb150a8d4c18330a967fedfe00fd073a5f0ac95b8b
        command:
          - /go/bin/che-machine-exec
          - '--url'
          - 127.0.0.1:3333
      attributes:
        app.kubernetes.io/component: machine-exec
        app.kubernetes.io/part-of: che-theia.eclipse.org
    - name: remote-runtime-injector
      container:
        image: >-
          quay.io/eclipse/che-theia-endpoint-runtime-binary@sha256:79df39dcc1bc37503a94303a25e1eabc985ce820d552d16a4b85ec444a7b2b1f
        env:
          - name: PLUGIN_REMOTE_ENDPOINT_EXECUTABLE
            value: /remote-endpoint/plugin-remote-endpoint
          - name: REMOTE_ENDPOINT_VOLUME_NAME
            value: remote-endpoint
        volumeMounts:
          - name: plugins
            path: /plugins
          - name: remote-endpoint
            path: /remote-endpoint
      attributes:
        app.kubernetes.io/component: remote-runtime-injector
        app.kubernetes.io/part-of: che-theia.eclipse.org
    - name: remote-endpoint
      volume:
        ephemeral: true
---
apiVersion: workspace.devfile.io/v1alpha2
kind: DevWorkspace
metadata:
  name: quarkus-quickstart
spec:
  started: true
  template:
    components:
      - name: tools
        container:
          image: registry.access.redhat.com/ubi8/openjdk-11
          command:
            - tail
          args:
            - sh
            - '-c'
            - ${PLUGIN_REMOTE_ENDPOINT_EXECUTABLE}
          env:
            - name: QUARKUS_HTTP_HOST
              value: 0.0.0.0
            - name: PLUGIN_REMOTE_ENDPOINT_EXECUTABLE
              value: /remote-endpoint/plugin-remote-endpoint
            - name: THEIA_PLUGINS
              value: local-dir:///plugins/sidecars/tools
          endpoints:
            - exposure: none
              name: debug
              protocol: tcp
              targetPort: 5005
            - exposure: public
              name: hello-greeting-endpoint
              protocol: http
              targetPort: 8080
              path: /hello/greeting/che-user
          volumeMounts:
            - name: m2
              path: /home/jboss/.m2
            - path: /remote-endpoint
              name: remote-endpoint
            - path: /plugins
              name: plugins
          memoryLimit: 4G
          mountSources: true
        attributes:
          che-theia.eclipse.org/vscode-extensions:
            - >-
              https://download.jboss.org/jbosstools/static/jdt.ls/stable/java-0.82.0-369.vsix
            - >-
              https://download.jboss.org/jbosstools/vscode/3rdparty/vscode-java-debug/vscode-java-debug-0.26.0.vsix
            - >-
              https://open-vsx.org/api/vscjava/vscode-java-test/0.28.1/file/vscjava.vscode-java-test-0.28.1.vsix
          che-theia.eclipse.org/vscode-preferences:
            java.server.launchMode: Standard
          app.kubernetes.io/name: tools
      - name: ubi-minimal
        container:
          image: registry.access.redhat.com/ubi8/ubi-minimal
          command:
            - tail
          args:
            - '-f'
            - /dev/null
          memoryLimit: 64M
          mountSources: true
      - name: gradle
        volume:
          size: 1G
      - name: m2
        volume:
          size: 1G
      - name: theia-ide-quarkus-quickstart
        plugin:
          kubernetes:
            name: theia-ide-quarkus-quickstart
    commands:
      - id: package
        exec:
          component: tools
          workingDir: ${PROJECTS_ROOT}/quarkus-quickstarts/getting-started
          commandLine: ./mvnw package
          group:
            kind: build
            isDefault: true
      - id: packagenative
        exec:
          label: Package Native
          component: tools
          workingDir: ${PROJECTS_ROOT}/quarkus-quickstarts/getting-started
          commandLine: >-
            ./mvnw package -Dnative -Dmaven.test.skip
            -Dquarkus.native.native-image-xmx=2G
          group:
            kind: build
      - id: startdev
        exec:
          label: Start Development mode (Hot reload + debug)
          component: tools
          workingDir: ${PROJECTS_ROOT}/quarkus-quickstarts/getting-started
          commandLine: ./mvnw compile quarkus:dev
          group:
            kind: run
            isDefault: true
      - id: startnative
        exec:
          label: Start Native
          component: ubi-minimal
          workingDir: ${PROJECTS_ROOT}/quarkus-quickstarts/getting-started/target
          commandLine: ./getting-started-1.0.0-SNAPSHOT-runner
          group:
            kind: run
    projects:
      - name: quarkus-quickstarts
        zip:
          location: 'http://:/resources/v2/quarkus-quickstarts.zip'
```

**Comment by svor** [comment]
@vinokurig thanks!
I think we need to replace images like `quay.io/eclipse/che-theia`, `quay.io/eclipse/che-theia-endpoint-runtime-binary`, `quay.io/eclipse/che-machine-exec` 
cc @nickboldt

**Comment by vinokurig** [comment]
@svor @nickboldt @benoitf 
> I think we need to replace images like quay.io/eclipse/che-theia, quay.io/eclipse/che-theia-endpoint-runtime-binary, quay.io/eclipse/che-machine-exec

I see three solutions here:
* Update the `@eclipse-che/che-theia-devworkspace-handler`, add a new parameter to be able to modify images
* Fork the `@eclipse-che/che-theia-devworkspace-handler` tool in the CRW repo and setup it with needed images.
* Update the `yaml` with correct images after it's generation.

**Comment by nickboldt** [comment]
updating yaml is likely easiest since we already do that in many places :) 

But having the handler able to be passed URLs would be a better longer-term solution for others who might want to build custom devfile/plugin registries without having to hack around with sed & yq

**Comment by svor** [comment]
+1 to update yamls in the script and I'll register an issue to extend the library

**Comment by nickboldt** [comment]
> +1 to update yamls in the script and I'll register an issue to extend the library
> 
> Here is the issue [eclipse/che#20967](https://github.com/eclipse/che/issues/20967)

Changing the registry won't help w/ CRW/whitelabelling, since the entire image is different...

- [ ]           quay.io/eclipse/che-theia@sha256:ebcc893364c233f4ebbaef61c6b2027c2aabecf05844f6151837dbb13193480d
- [ ]           quay.io/eclipse/che-machine-exec@sha256:2ecefe3a93c1ce35b0d56beb150a8d4c18330a967fedfe00fd073a5f0ac95b8b
- [ ]           quay.io/eclipse/che-theia-endpoint-runtime-binary@sha256:79df39dcc1bc37503a94303a25e1eabc985ce820d552d16a4b85ec444a7b2b1f

Also not sure how these will work in airgap... are we caching them inside the container somehow?

```
che-theia.eclipse.org/vscode-extensions:
            - >-
              https://download.jboss.org/jbosstools/static/jdt.ls/stable/java-0.82.0-369.vsix
            - >-
              https://download.jboss.org/jbosstools/vscode/3rdparty/vscode-java-debug/vscode-java-debug-0.26.0.vsix
            - >-
              https://open-vsx.org/api/vscjava/vscode-java-test/0.28.1/file/vscjava.vscode-java-test-0.28.1.vsix
```

**Review by nickboldt — CHANGES_REQUESTED** [review]
need updates to [build/scripts/sync.sh](https://github.com/redhat-developer/codeready-workspaces-images/blob/crw-2-rhel-8/codeready-workspaces-devfileregistry/build/scripts/sync.sh) script as noted in https://github.com/redhat-developer/codeready-workspaces/pull/652#issuecomment-999686702

**Comment by svor** [comment]
> Also not sure how these will work in airgap... are we caching them inside the container somehow?

@nickboldt good catch i think it won't work, it should be replaced to reference local resources like 
`relative:extension/resources/download_jboss_org/jbosstools/static/jdt_ls/stable/java-0.75.0-60.vsix`

I see that factory flow uses relative path, we need to investigate how to do the same with the handler library

**Comment by vinokurig** [comment]
@nickboldt 
Instead of updating the `sync.sh` script I added `--plugin-registry-url:https://crw-plugin-registry-main.surge.sh/v3/` key to the `@eclipse-che/che-theia-devworkspace-handler` and now the generated template has proper images:
```
apiVersion: workspace.devfile.io/v1alpha2
kind: DevWorkspaceTemplate
metadata:
  name: theia-ide-quarkus-quickstart
spec:
  commands:
    - id: init-container-command
      apply:
        component: remote-runtime-injector
  events:
    preStart:
      - init-container-command
  components:
    - name: theia-ide
      container:
        image: registry.redhat.io/codeready-workspaces/theia-rhel8:2.15
        env:
          - name: THEIA_PLUGINS
            value: local-dir:///plugins
          - name: HOSTED_PLUGIN_HOSTNAME
            value: 0.0.0.0
          - name: HOSTED_PLUGIN_PORT
            value: '3130'
          - name: THEIA_HOST
            value: 127.0.0.1
        volumeMounts:
          - name: plugins
            path: /plugins
          - name: theia-local
            path: /home/theia/.theia
        mountSources: true
        memoryLimit: 512M
        cpuLimit: 1500m
        cpuRequest: 100m
        endpoints:
          - name: theia
            attributes:
              type: main
              cookiesAuthEnabled: true
              discoverable: false
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: webviews
            attributes:
              type: webview
              cookiesAuthEnabled: true
              discoverable: false
              unique: true
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: mini-browser
            attributes:
              type: mini-browser
              cookiesAuthEnabled: true
              discoverable: false
              unique: true
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: theia-dev
            attributes:
              type: ide-dev
              discoverable: false
              urlRewriteSupported: true
            targetPort: 3130
            exposure: public
            protocol: http
          - name: theia-redirect-1
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13131
            exposure: public
            protocol: http
          - name: theia-redirect-2
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13132
            exposure: public
            protocol: http
          - name: theia-redirect-3
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13133
            exposure: public
            protocol: http
          - name: terminal
            attributes:
              type: collocated-terminal
              discoverable: false
              cookiesAuthEnabled: true
              urlRewriteSupported: true
            targetPort: 3333
            exposure: public
            secure: false
            protocol: wss
      attributes: {}
    - name: plugins
      volume: {}
    - name: theia-local
      volume: {}
    - name: che-machine-exec
      container:
        image: registry.redhat.io/codeready-workspaces/machineexec-rhel8:2.15
        command:
          - /go/bin/che-machine-exec
          - '--url'
          - 127.0.0.1:3333
          - '--idle-timeout'
          - 15m
        memoryLimit: 128Mi
        memoryRequest: 32Mi
        cpuLimit: 500m
        cpuRequest: 30m
      attributes: {}
    - name: remote-runtime-injector
      container:
        image: registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8:2.15
        env:
          - name: PLUGIN_REMOTE_ENDPOINT_EXECUTABLE
            value: /remote-endpoint/plugin-remote-endpoint
          - name: REMOTE_ENDPOINT_VOLUME_NAME
            value: remote-endpoint
        volumeMounts:
          - name: plugins
            path: /plugins
          - name: remote-endpoint
            path: /remote-endpoint
        memoryLimit: 128Mi
        memoryRequest: 32Mi
        cpuLimit: 500m
        cpuRequest: 30m
    - name: remote-endpoint
      volume:
        ephemeral: true
---
apiVersion: workspace.devfile.io/v1alpha2
kind: DevWorkspace
metadata:
  name: quarkus-quickstart
spec:
  started: true
  template:
    components:
      - name: tools
        container:
          image: registry.access.redhat.com/ubi8/openjdk-11
          command:
            - tail
          args:
            - sh
            - '-c'
            - ${PLUGIN_REMOTE_ENDPOINT_EXECUTABLE}
          env:
            - name: QUARKUS_HTTP_HOST
              value: 0.0.0.0
            - name: PLUGIN_REMOTE_ENDPOINT_EXECUTABLE
              value: /remote-endpoint/plugin-remote-endpoint
            - name: THEIA_PLUGINS
              value: local-dir:///plugins/sidecars/tools
          endpoints:
            - exposure: none
              name: debug
              protocol: tcp
              targetPort: 5005
            - exposure: public
              name: hello-greeting-endpoint
              protocol: http
              targetPort: 8080
              path: /hello/greeting/che-user
          volumeMounts:
            - name: m2
              path: /home/jboss/.m2
            - name: gradle
              path: /home/jboss/.gradle
            - path: /remote-endpoint
              name: remote-endpoint
            - path: /plugins
              name: plugins
          memoryLimit: 4G
          mountSources: true
        attributes:
          che-theia.eclipse.org/vscode-extensions:
            - >-
              relative:extension/resources/download_jboss_org/jbosstools/static/jdt_ls/stable/java-0.75.0-60.vsix
            - >-
              relative:extension/resources/download_jboss_org/jbosstools/vscode/3rdparty/vscode-java-debug/vscode-java-debug-0.26.0.vsix
          che-theia.eclipse.org/vscode-preferences:
            java.server.launchMode: Standard
          app.kubernetes.io/name: tools
      - name: ubi-minimal
        container:
          image: registry.access.redhat.com/ubi8/ubi-minimal
          command:
            - tail
          args:
            - '-f'
            - /dev/null
          memoryLimit: 64M
          mountSources: true
      - name: gradle
        volume:
          size: 1G
      - name: m2
        volume:
          size: 1G
      - name: theia-ide-quarkus-quickstart
        plugin:
          kubernetes:
            name: theia-ide-quarkus-quickstart
    commands:
      - id: package
        exec:
          component: tools
          workingDir: ${PROJECTS_ROOT}/quarkus-quickstarts/getting-started
          commandLine: ./mvnw package
          group:
            kind: build
            isDefault: true
      - id: packagenative
        exec:
          label: Package Native
          component: tools
          workingDir: ${PROJECTS_ROOT}/quarkus-quickstarts/getting-started
          commandLine: >-
            ./mvnw package -Dnative -Dmaven.test.skip
            -Dquarkus.native.native-image-xmx=2G
          group:
            kind: build
      - id: startdev
        exec:
          label: Start Development mode (Hot reload + debug)
          component: tools
          workingDir: ${PROJECTS_ROOT}/quarkus-quickstarts/getting-started
          commandLine: ./mvnw compile quarkus:dev
          group:
            kind: run
            isDefault: true
      - id: startnative
        exec:
          label: Start Native
          component: ubi-minimal
          workingDir: ${PROJECTS_ROOT}/quarkus-quickstarts/getting-started/target
          commandLine: ./getting-started-1.0.0-SNAPSHOT-runner
          group:
            kind: run
    projects:
      - name: quarkus-quickstarts
        zip:
          location: 'http://:/resources/v2/quarkus-quickstarts.zip'
```
I still not sure about using https://crw-plugin-registry-main.surge.sh/v3/ as a stable plugin registry url

**Comment by svor** [comment]
@vinokurig 
```
    projects:
      - name: quarkus-quickstarts
        zip:
          location: 'http://:/resources/v2/quarkus-quickstarts.zip'
```
I see that the location contains `://:/` , should we improve something?

**Comment by vinokurig** [comment]
@svor 
> I see that the location contains ://:/ , should we improve something?

This is because I ran the container on my local machine on localhost: `docker run -it --rm -p 8080:8080 quay.io/crw/devfileregistry-rhel8:next`

**Comment by nickboldt** [comment]
surge.sh won't work for offline/airgap. 

How is https://crw-plugin-registry-main.surge.sh/v3/ updated? Is it based on crw-2-rhel-8 branch? or crw-2.14 branch? how can different versions of CRW use the same registry?

File: dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+set -e
+
+# shellcheck disable=SC1091
+source ./clone_and_zip.sh
+
+npm install -g @eclipse-che/che-theia-devworkspace-handler@0.0.1-1640160609
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh:16]: Hardcoding the version here means it'll likely get out of date... can we move this to a central location so it's kept up to date?

File: dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 Red Hat, Inc.
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh:3]: new script for 2022

File: dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh (hunk 2 above)
Comment (nickboldt) [reply dependencies/che-devfile-registry/build/scripts/generate_devworkspace_templates.sh:16]: Case in point, https://www.npmjs.com/package/@eclipse-che/che-theia-devworkspace-handler has 0.0.1-1640160609 as released 19 days ago, 6 releases ago.

**Comment by svor** [comment]
> surge.sh won't work for offline/airgap.
> 
> How is https://crw-plugin-registry-main.surge.sh/v3/ updated? Is it based on crw-2-rhel-8 branch? or crw-2.14 branch? how can different versions of CRW use the same registry?

It generates and updates by GitHub action https://github.com/redhat-developer/codeready-workspaces/actions/workflows/plugin-registry-build-publish-content.yaml from crw-2-rhel-8 branch. 
So, we should find another way how to update images and vsix paths

**Review by nickboldt — APPROVED** [review]

File: dependencies/che-devfile-registry/versions.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "che-theia-devworkspace-handler": "0.0.1-1643111178"
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/versions.json:2]: thanks for extracting this out but the problem remains... how do we keep it up to date? How are you handling this in upstream?

Comment (svor) [reply dependencies/che-devfile-registry/versions.json:2]: In upstream we update the version by hands - it's not automated. @vinokurig can correct me if I'm wrong

Comment (vinokurig) [reply dependencies/che-devfile-registry/versions.json:2]: I've filed an issue to sync it's version with CHE. When it is done, we can add it to automation update here: https://github.com/redhat-developer/codeready-workspaces/blob/crw-2-rhel-8/product/updateVersionAndRegistryTags.sh#L128-L151

Comment (nickboldt) [reply dependencies/che-devfile-registry/versions.json:2]: ... link to the issue?

Comment (vinokurig) [reply dependencies/che-devfile-registry/versions.json:2]: > ... link to the issue?

https://github.com/eclipse/che/issues/21104

**Pull request merged**